agent/osv_output_handler.py

```diff
@@ -79,48 +87,87 @@ def parse_results(output: str, api_key: str | None = None) -> Iterator[Vulnerabi
-                )
-                recommendation = f"We recommend updating `{package_name}` to the latest available version."
-                yield Vulnerability(
-                    entry=kb.Entry(
-                        title=f"Use of Outdated Vulnerable Component: "
-                        f"{package_name}@{package_version}: {', '.join(cve_ids)}",
-                        risk_rating=risk_rating.upper(),
-                        short_description=vuln.get("summary", ""),
-                        description=description,
-                        references=build_references(vuln.get("references", [])),
-                        security_issue=True,
-                        privacy_issue=False,
-                        has_public_exploit=False,
-                        targeted_by_malware=False,
-                        targeted_by_ransomware=False,
-                        targeted_by_nation_state=False,
-                        recommendation=recommendation,
-                    ),
-                    technical_detail=f"{vuln.get('details')} \n#### CVEs:\n {cve_list_details}",
-                    risk_rating=RISK_RATING_MAPPING[risk_rating.upper()],
+            parsed_vulns.extend(
+                parse_vulnerabilities(
+                    output=package,
+                    api_key=api_key,
+                    file_type=file_type,
+                    file_name=file_name,
                 )
+            )
+
+    return parsed_vulns
+
+
+def parse_vulnerabilities(
+    output: dict[str, Any],
+    package_name: str | None = None,
+    package_version: str | None = None,
+    api_key: str | None = None,
+    file_type: str | None = None,
+    file_name: str | None = None,
+) -> list[VulnData]:
+    """Parse the OSV API response to extract vulnerabilities.
+    Args:
+        output: The API response json.
+        package_name: The package name.
+        package_version: The package version.
+        api_key: The NVD API key.
+    Returns:
+        Parsed output.
+    """
+    try:
+        vulnerabilities = output.get("vulns") or output.get("vulnerabilities") or []
+        package_name = output.get("package", {}).get("name") or package_name
+        package_version = output.get("package", {}).get("version") or package_version
+        parsed_vulns = []
+        for vulnerability in vulnerabilities:
+            risk = vulnerability.get("database_specific", {}).get("severity")
+            filtered_cves = [
+                alias for alias in vulnerability.get("aliases", []) if "CVE" in alias
+            ]
+            if risk is None:
+                risk_ratings = []
+                for cve in filtered_cves:
+                    cve_data = cve_service_api.get_cve_data_from_api(cve, api_key)
+                    risk_ratings.append(cve_data.risk)
+                risk = calculate_risk_rating(risk_ratings)
+            elif risk == "MODERATE":
+                risk = "MEDIUM"
```

Comment:
Why not add this to calculate_risk_rating

Reply:
we calculate the risk when its None. This check is just to match our risk levels `["CRITICAL", "HIGH", "MEDIUM", "LOW", "POTENTIALLY"]` sinnce we use medium but osv return MODERATE